src/main/io/gps.c

```diff
@@ -2613,6 +2613,10 @@ void onGpsNewData(void)
     GPS_calculateDistanceAndDirectionToHome();
     if (ARMING_FLAG(ARMED)) {
         GPS_calculateDistanceFlown(false);
+
+        if (!STATE(GPS_FIX_HOME)) {
+            GPS_reset_home_position();
+        }
```

Comment:
Now we have the "gps_rescue_allow_arming_without_fix" that has ON/OFF values. I use in ON and I'm conscious of it because if I take off there is an OSD message that informs of it (but with moderns GPS this usually not happens because they acquire sats fast).
Maybe to mitigate the risks is good to have a new value here "ON/OFF/ACQUIRE_HOME_IN_FLIGHT" or similar, or create a new CLI parameter.
And add an OSD warning when this occurs, to let the user know this circumstance.

Reply:
In this case, you will also have to make changes to the configurator, as well as make some kind of pop-up hint informing that one of the modes will lead to the drone crashing in flight, the user will not choose this)
I also wouldn’t want to make any significant changes since I know very little “C” and the architecture of these projects.

You can make an ON OFF setting only in the CLI specifically for this mode, without displaying it in the GUI Configurator. And make it enabled by default.